### PR TITLE
Feat/hot 005/setup tailwind navbar

### PR DESCRIPTION
**Pull request by katporks**
This PR introduces the configuration of Tailwind CSS and Alpine.js to construct the header component. Alpine.js variables, defined within `header.html`, are accessible in nested templates due to their initialization in the parent component `header`. It's important to note that Alpine.js operates entirely on the client-side, in the browser, and interacts with the HTML after Django's server-side rendering is complete.

The header component includes the following variables:

- `isMenuOpen`: This variable indicates the state of the mobile hamburger menu, determining whether the mobile navigation should be displayed.
- `isSearchClicked`: This variable controls the visibility of the search modal upon clicking the search icon.

This PR lays the groundwork for the header and navigation. In a subsequent PR, once the site map's links are finalized, I plan to incorporate submenus for the parent links in both the desktop and mobile navigation.

Desktop header:
Desktop header

Mobile header:
Mobile header

### CODE REVIEW

**Comment by JoltCode** [comment]
Thanks Kat! Will take a look too 👍

File: frontend/tailwind.config.js [hunk 1]
```diff
@@ -1,12 +1,64 @@
...
+          "hot-off-white": "#F0EFEF",
+          "hot-white": "#FFFFFF",
+          "hot-black": "#000000",
+        },
+        borderRadius: {
+          8: "8px",
+        },
+        borderWidth: {
+          "2pt": "2.66667px",
+        },
```
Comment (JoltCode) [inline frontend/tailwind.config.js:57]: Why are we using pt?

File: hot_osm/templates/components/branded_elements/button.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<button class="font-medium font-barlow h-hot-4 text-base-18 bg-hot-red text-hot-white px-hot-3 rounded-8 border-2pt border-transparent hover:bg-hot-off-white hover:text-hot-red focus:bg-transparent focus:text-hot-red focus:outline-none nav-breakpoint:ml-3 {{ classes }}">
```
Comment (JoltCode) [inline hot_osm/templates/components/branded_elements/button.html:1]: Why do we have a transparent border here? - also related to the comment on using pt :)

File: hot_osm/templates/components/branded_elements/link.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<a href="{{ link }}"
+   class="hover:underline block font-medium font-barlow text-hot-black {{ classes }}">
+    {{ text }}
+    {% if parent|default:False %}
+        <span class="absolute translate-y-[-15%] translate-x-1">⌄</span>
```
Comment (JoltCode) [inline hot_osm/templates/components/branded_elements/link.html:5]: Don't use absolute here, use padding instead - also with the translate, you can make it negative in tailwind by appending a `-` at the start 👍

File: hot_osm/templates/components/mobile/mobile_menu_button.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<button type="button"
+        @click="isMenuOpen = !isMenuOpen"
+        class="relative inline-flex items-center justify-center rounded-md p-2 text-gray-500 hover:bg-gray-100 hover:text-gray-500 focus:outline-none focus:ring-inset nav-breakpoint:hidden"
+        aria-controls="mobile-menu"
+        aria-expanded="false">
+    <span class="absolute -inset-0.5"></span>
```
Comment (JoltCode) [inline hot_osm/templates/components/mobile/mobile_menu_button.html:6]: What's this for?

File: hot_osm/templates/components/mobile/mobile_menu_link.html [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<a href="{{ link }}"
+   class="block font-medium font-barlow text-hot-black py-2 pl-3 pr-4 border-l-4 text-base {% if selected|default:False %}border-hot-red bg-hot-light-grey text-hot-red {% else %} border-transparent hover:border-gray-300 hover:bg-gray-50 hover:text-gray-800 {% endif %}{{ classes }}">
+    {{ text }}
+    {% if parent|default:False %}
+        <span class="absolute translate-y-[-15%] translate-x-1">⌄</span>
```
Comment (JoltCode) [inline hot_osm/templates/components/mobile/mobile_menu_link.html:5]: And an aria text label here :)

File: hot_osm/templates/components/mobile/mobile_navigation.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<div x-show="isMenuOpen"
+     @click.outside="isMenuOpen = false"
+     class="font-barlow nav-breakpoint:hidden"
+     id="mobile-menu">
+    <nav class="space-y-1 pb-3 pt-2 font-medium text-base-18"
```
Comment (JoltCode) [inline hot_osm/templates/components/mobile/mobile_navigation.html:5]: Use `gap` - it's cleaner, and doesn't introduce margins on children, rather it adds gaps between children at the flexbox parent level 👍

File: hot_osm/templates/components/mobile/mobile_navigation.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Who We Are" selected=True parent=True %}</li>
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Our Work" parent=True %}</li>
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Open Mapping Hubs" parent=True %}</li>
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Tools &amp; Resources" parent=True %}</li>
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Get Involved" parent=True %}</li>
+            <li>{% include "../branded_elements/button.html" with text="Start Mapping" classes="my-2 ml-3 mr-4" %}</li>
+        </ul>
+    </nav>
+    <nav class="border-t border-b border-gray-200 py-2">
+        <ul class="space-y-0.5 list-none pl-0">
```
Comment (JoltCode) [inline hot_osm/templates/components/mobile/mobile_navigation.html:17]: ^ - gap :)

File: hot_osm/templates/components/nav/primary_desktop_nav.html [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<div class="hidden nav-breakpoint:block">
+    <nav>
+        <ul class="flex text-base-18 p-0 flex-row space-x-4 mt-0 border-0"
```
Comment (JoltCode) [inline hot_osm/templates/components/nav/primary_desktop_nav.html:3]: ^ - gap :)

File: hot_osm/templates/components/branded_elements/social_icons.html [hunk 9]
```diff
@@ -0,0 +1,47 @@
+<ul class="flex flex-shrink-0 h-hot-2 space-x-1.5">
```
Comment (JoltCode) [inline hot_osm/templates/components/branded_elements/social_icons.html:1]: ^ - gap :)

**Review by JoltCode — CHANGES_REQUESTED** [review]
Great work! Just a few CSS changes I'd suggest 👍 - I haven't actually looked at the server in my browser, I don't have the tasking manager website set up.

**Review by spwoodcock — APPROVED** [review]
Looks great - nice use of Tailwind / CSS!

@JoltCode may have a better idea how to fix the custom spacing and breakpoint issue 😄 

The great thing is that Alpine.js works nicely with Web Components, so hopefully integrating them shouldn't be too big of a refactor:
- Any logic encapsulated in the Web Component can replace the alpine.js equivalent.
- Anything missing from the Web Component logic can be supplemented by alpine.js easily.

File: frontend/tailwind.config.js [hunk 10]
```diff
@@ -1,12 +1,58 @@
...
-  purge: ['./**/*.html', './**/*.js',],
-  content: ["*/templates/**/*.html"],
+  future: {},
+  content: ["../*/templates/*.{html,js}", "../*/templates/**/*.{html,js}"],
+  darkMode: "media",
   theme: {
-    extend: {},
+    extend: {
+      screens: {
+          "nav-breakpoint": "1050px", // prevents nav  links from wrapping into two lines
```
Comment (JoltCode) [inline frontend/tailwind.config.js:10]: Could we use `lg:` instead? It's pretty close to the default tailwind `1024px` breakpoint :)

File: hot_osm/templates/components/branded_elements/social_icons.html [hunk 11]
```diff
@@ -0,0 +1,56 @@
+<ul class="flex flex-shrink-0 h-hot-2 gap-x-1">
+    <li>
+        <a href="https://www.facebook.com/hotosm" aria-label="Facebook">
```
Comment (JoltCode) [inline hot_osm/templates/components/branded_elements/social_icons.html:3]: Good stuff on all these aria labels! 😄

File: hot_osm/templates/components/mobile/mobile_menu_link.html [hunk 12]
```diff
@@ -0,0 +1,17 @@
+<a href="{{ link }}"
+   class="block font-medium font-barlow text-hot-black py-2 pl-3 pr-4 border-l-4 text-base {% if selected|default:False %}border-hot-red bg-hot-light-grey text-hot-red {% else %} border-transparent hover:border-gray-300 hover:bg-gray-50 hover:text-gray-800 {% endif %}{{ classes }}"
```
Comment (JoltCode) [inline hot_osm/templates/components/mobile/mobile_menu_link.html:2]: Does the `hover:border-gray-300` need to be changed to a HOT colour?

File: hot_osm/templates/components/mobile/mobile_navigation.html [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+        <ul class="list-none pl-0">
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Who We Are" selected=True parent=True %}</li>
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Our Work" parent=True %}</li>
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Open Mapping Hubs" parent=True %}</li>
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Tools &amp; Resources" parent=True %}</li>
+            <li>{% include "../mobile/mobile_menu_link.html" with link="#" text="Get Involved" parent=True %}</li>
+            <li>{% include "../branded_elements/button.html" with text="Start Mapping" classes="my-2 ml-3 mr-4" %}</li>
+        </ul>
+    </nav>
+    <nav class="border-t border-b border-gray-200 py-2">
```
Comment (JoltCode) [inline hot_osm/templates/components/mobile/mobile_navigation.html:16]: Again just checking if we're good to use the tailwindcss colours, not the HOT colours for these borders 👍

**Comment by JoltCode** [comment]
> Looks great - nice use of Tailwind / CSS!
> 
> @JoltCode may have a better idea how to fix the custom spacing and breakpoint issue 😄
> 
> The great thing is that Alpine.js works nicely with Web Components, so hopefully integrating them shouldn't be too big of a refactor:
> 
> * Any logic encapsulated in the Web Component can replace the alpine.js equivalent.
> * Anything missing from the Web Component logic can be supplemented by alpine.js easily.

Sure thing! What're the specific issues we're facing? - Feel free to PM about this! 😄

**Pull request merged**